PublicMapping/districtbuilder
# Private / published maps

## Overview

This PR expands the functionality of the share menu to allow controling the visiblity of a map, which can now be:
 - private (the default for new maps)
 - share with link (the current behavior)
 - published (effectively the same as visible, but in a later feature we may list these publicly somewhere)

### Checklist


### Demo

![image](https://user-images.githubusercontent.com/4432106/105760167-5fc06000-5f1f-11eb-8b6f-38c898ec5551.png)


### Notes

I made the default visibility for all existing maps "share with link" since that matches existing behavior and I didn't want to mess up anyone's already shared maps, but the default for _new_ maps is private as requested in the issue.

## Testing Instructions

- `scripts/migration`
- Existing maps should default to a visibility of Shared with link. New maps should default to Private.
- Ensure the backend appropriately returns 404s for private maps w/o authentication or when logged in as another user

Closes #533
